bciers/apps/reporting/src/app/components/attachments/AttachmentsForm.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+import { multiStepHeaderSteps } from "../taskList/multiStepHeaderConfig";
+import { HasReportVersion } from "../../utils/defaultPageFactory";
+import postAttachments from "@reporting/src/app/utils/postAttachments";
+import AttachmentElement from "./AttachmentElement";
+import { useState } from "react";
+import { UploadedAttachment } from "./types";
+import { getDictFromAttachmentArray } from "./AttachmentsPage";
+import { TaskListElement } from "@bciers/components/navigation/reportingTaskList/types";
+import MultiStepWrapperWithTaskList from "@bciers/components/form/MultiStepWrapperWithTaskList";
+
+interface Props extends HasReportVersion {
+  initialUploadedAttachments: {
+    [attachment_type: string]: UploadedAttachment;
+  };
+  taskListElements: TaskListElement[];
+}
+
+const AttachmentsForm: React.FC<Props> = ({
+  version_id,
+  taskListElements,
+  initialUploadedAttachments,
+}) => {
+  const [pendingUploadFiles, setPendingUploadFiles] = useState<{
+    [fileType: string]: File;
+  }>({});
+  const [uploadedAttachments, setUploadedAttachments] = useState(
+    initialUploadedAttachments,
+  );
+
+  const handleChange = (fileType: string, file: File | undefined) => {
+    if (file) {
+      setPendingUploadFiles({ ...pendingUploadFiles, [fileType]: file });
+    } else {
+      const { [fileType]: removed, ...prunedFiles } = pendingUploadFiles;
+      setPendingUploadFiles(prunedFiles);
+    }
+  };
+
+  const handleSubmit = async () => {
+    // Nothing to submit
+    if (Object.keys(pendingUploadFiles).length === 0) return;
+
+    const formData = new FormData();
+
+    for (const [fileType, file] of Object.entries(pendingUploadFiles)) {
+      formData.append("files", file);
+      formData.append("file_types", fileType);
+    }
+
+    const response = await postAttachments(version_id, formData);
+    const newUploadedAttachments = getDictFromAttachmentArray(response);
+
+    setUploadedAttachments(newUploadedAttachments);
+    setPendingUploadFiles({});
```

Comment:
Perhaps, add the router push functionality of the continue part of "Save and continue"?

Something like:
```
 const queryString = serializeSearchParams(useSearchParams());
 const saveAndContinueUrl = `/reports/${version_id}/final-review${queryString}`;

const handleSubmit = async () => {
...
   if (response?.error) {
      setError(response.error);
      return;
    } else {
      setError(undefined);
    }
    router.push(saveAndContinueUrl);

}
```